include/zephyr/devicetree.h

```diff
@@ -917,6 +917,226 @@
 #define DT_STRING_TOKEN(node_id, prop) \
 	DT_CAT4(node_id, _P_, prop, _STRING_TOKEN)
 
+/**
+ * @brief Get a string property's value as a token.
+ *
+ * This removes "the quotes" from a string property's value,
+ * converting any non-alphanumeric characters to underscores. This can
+ * be useful, for example, when programmatically using the value to
+ * form a C variable or code.
```

Comment:
Please revisit the docstring. It looks like things were copy/pasted from elsewhere that don't accurately reflect the behavior of these macros.